components/ActivityDetails.js

```diff
@@ -0,0 +1,86 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import { activities } from "@/lib/activities"; 
+import Link from "next/link";
+import Image from "next/image";
+
+export default function ActivityDetails() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const activity = activities.find(activity => activity.id === id);
+
+  if (!activity) return <p>Loading...</p>;
```

Comment:
The component shouldn't be responsible for checking the current route. It's a good practice to keep components as dumb as possible. Instead move the router logic to the page `pages/activity/[id].js` and pass the found activity to the component as a prop.